src/components/Tables/DomainTable/DomainTable.tsx

```diff
@@ -203,56 +202,69 @@ const DomainTable: React.FC<DomainTableProps> = ({
 					);
 				}
 
+				const loaded = data;
+
 				loaded.push({
 					domain: domain,
 					metadata: metadata,
 					bids: filteredBids || bids || [],
 				});
 
-				if (completed === count) {
-					setLoadedDomains(loaded);
-					setHasMetadataLoaded(true);
-					setIsLoading(false);
-					if (onLoad) {
-						onLoad();
-					}
-				}
+				setData(loaded);
+
+
 			} catch (e) {}
 		};
 
-		for (let i = 0; i < domains.length; i++) {
-			if (!domains[i].metadata) continue;
-			count++;
-			getData(domains[i]);
-		}
-
-		if (!count) {
-			setHasMetadataLoaded(true);
-			setIsLoading(false);
+		if (isSubscribed === false && domains.length > 0) {
+			//if finished before call to all asyncs fetches, and the domains exist
+			setIsSubscribed(true);
+			for (let i = 0; i < domains.length; i++) {
+				if (!domains[i].metadata) continue;
+				getData(domains[i]);
+			}
 		}
 	}, [domains]);
 
 	useEffect(() => {
 		if (!isLoading && onLoad) onLoad();
 	}, [isLoading]);
 
+	/////////////////////
+	///// Functions /////
+	/////////////////////
+
+	const feed = () => {
+		let cachedData = tableData; //last data feeded
+
+		let lengthDiff = data.length - cachedData.length
+		let feetIteration =  lengthDiff < chunkLength ? lengthDiff : chunkLength //how many data we need to feed, based on the fetched data we have
+		
+		for (
+			let i = cachedData.length;
+			i < feetIteration;
+			i++ //adds new chunk of 5 domains
```

Comment:
Remove comments like this, the 5 refers to the setting of `chunkLength` but could be changed and then this comment is outdated and confusing.

Reply:
Also why is it only being added in chunks?

Reply:
Changed to only load one first chunk, right now first 5 domains. Brett will call feed() later and in each call it adds one domain more to the table, so he can load on demand instead of chunks